packages/mobile/src/components/audio/GoogleCast.tsx

```diff
@@ -70,12 +104,20 @@ export const useChromecast = () => {
     }
   }, [castState, dispatch])
 
+  // Ensure that the progress gets reset to 0
+  // when a new track is played
+  useEffect(() => {
+    if (prevTrack && prevTrack !== track) {
+      global.progress.currentTime = 0
+    }
+  }, [prevTrack, track])
+
   // Load media when the cast connects
   useEffect(() => {
     if (castState === CastState.CONNECTED) {
       loadCast(track, global.progress.currentTime ?? 0)
     }
-  }, [loadCast, track, castState])
+  }, [loadCast, track, prevTrack, castState])
```

Comment:
does this work if you replay the same song? maybe it should be using play counter

Reply:
Ah, no it does not. Good idea, I think counter is the move